src/manifest.c

```diff
@@ -675,19 +675,21 @@ char *file_type_to_string(struct file *file)
 	return type;
 }
 
+/* Calculate the contentsize for the manifest based on file sizes.
+ * This should calculate the files uniquely included in this manifest, but none
+ * of its submanifests, which will allow calculation of all bundles on a system
+ * by adding all manifest->contentsizes of installed bundles.
+ */
```

Comment:
Altering the manifest to add the size of each file has an additional advantage, in that "swupd verify --fast" could check the size (it is already doing a stat call so the only extra time is converting the file size from ascii in the manifest). It could also be used whilst trying to work out if an update can fit on the disk.